packages/repository/src/__tests__/unit/decorator/relation.decorator.unit.ts

```diff
@@ -198,6 +198,55 @@ describe('relation decorator', () => {
         },
       });
     });
+
+    it('accepts additional property metadata', () => {
+      @model()
+      class AddressBook extends Entity {
+        @property({
+          id: true,
+          length: 36,
+          postgresql: {
+            dataType: 'uuid',
+          },
+        })
+        id: string;
+      }
+      @model()
+      class Address extends Entity {
+        id: string;
+        @belongsTo(
+          () => AddressBook,
+          {},
+          {
+            length: 36,
+            postgresql: {
+              dataType: 'uuid',
+            },
+          },
+        )
+        addressBookId: string;
```

Comment:
I'm not very happy that we can have to mix the property metadata with relations.

Ideally, `@belongsTo` should be applied to the `addressBook` relational property and `@property` to the `addressBookId`. Something like:
```ts
      @model()
      class Address extends Entity {
        id: string;
        @belongsTo(
          () => AddressBook,
          {},
        )
        addressBook?: AddressBook;

        @property({
            length: 36,
            postgresql: {
              dataType: 'uuid',
            },
          })
        addressBookId: string;
```

WDYT?

Reply:
What will the `address` table look like in the database? Specifically what is the column type of `addressBook`

Another approach might be to allow multiple decorators on a property, with the constraint that the decorator keys must be unique.

Reply:
I think that using two different properties in a model to define a relation is redundant.
In my opinion, the property `addressBook?: AddressBook;` @raymondfeng suggested should be also the one internally creating the column `addressBookId` in the DB. This way it can get automatically configured to comply with the PK it references.
In addition, property decorator (`@belongsTo()`) may also support additional configuration for the FK column in the DB (`addressBookId`).

Reply:
> ```ts
>       @model()
>       class Address extends Entity {
>         id: string;
>         @belongsTo(
>           () => AddressBook,
>           {},
>         )
>         addressBook?: AddressBook;
> 
>         @property({
>             length: 36,
>             postgresql: {
>               dataType: 'uuid',
>             },
>           })
>         addressBookId: string;
> ```

@raymondfeng I really like this approach as it is clearly seperates concerns and gives the developer more control of just reading the id or loading the related object (with subquery to the database).

Should this work already with the latest release or are changes needed to make this possible? (I tried this approach on my application and got an exception "500 Error: Invalid type for property type")

Reply:
When I run the code it creates a text column for addressBook. This seems confusing to have a `AddressBook` table and `AddressBook` column with the same information. I agree with @orshlom that two properties are redundant and maybe the `@belongsTo` property can automatically create an `addressBookId` property to link to the primary key.

Also I'm wondering what the API for the user will be when they call the `Address` endpoint.  Will the user be able to create an Address and AddressBook in a single POST. How will they add a reference to an existing addressBook?

@derdeka The issue you are running into might be related to `length`. You can move it to `dataLength` within `postgresql`, but when I do that I'm getting an error `type modifier is not allowed for type "uuid"`

Reply:
As I understood, as `belongsTo` decorator uses `property` decorator inside the code, this means that on every `belongsTo` decorator call - new property will be created, and then `belongsTo` creating the relation for the same.

https://github.com/strongloop/loopback-next/blob/4cee896e7f7b2b503b118e560c31036626587f47/packages/repository/src/relations/belongs-to/belongs-to.decorator.ts#L19-L56

For the suggested approach of splitting `belongsTo` and `property` decorators for defining the relation - this doesn't work, because it is trying to create the same property 2nd time.
Could we somehow map property which is created/defined in `belongsTo` to the property created with `property` decorator?
Or could we have creating property in `belongsTo` only when it is not created yet?
What do you think? How this could be solved?